### PR TITLE
[FASTLY] Add fastly completions with args

### PR DESCRIPTION
**Pull request by kailan**
generated with updates from https://github.com/kailan/figly/commit/0e9c2f2a8513e7f1e909fea6f6bc739200be071d

### CODE REVIEW

**Review by mattschrage — CHANGES_REQUESTED** [review]
One final tweak!

**Review by mattschrage — APPROVED** [review]
This looks great! 🎉

File: dev/fastly.ts [hunk 1]
```diff
@@ -0,0 +1,12449 @@
...
+        },
+      ],
+      options: [],
+    },
+  ],
+  options: [
+    {
+      name: "--help",
+      description: "Show context-sensitive help.",
+      args: [
```
Comment (falky97) [inline dev/fastly.ts:12424]: Hey - the spec looks good. I think the only thing that strikes me a little strange is why do--verbose and --help  have arguments? I am guessing they should not. Is this potentially a problem with the underlying parser? Would you be able to check this for us?

File: dev/fastly.ts [hunk 2]
```diff
@@ -0,0 +1,12449 @@
...
+                },
+              ],
+            },
+          ],
+        },
+      ],
+      options: [],
+    },
+  ],
+  options: [
```
Comment (falky97) [inline dev/fastly.ts:12420]: Also, reading the docs, these seem like global flags. Adding them at the top level doesn't automatically add them to all the relevant subcommands unfortunately. One option is your parser adds these options to every single command. Another option is we set a setting in the completino spec called something like "globalFlags" and we add them in from Fig's end for you. I presume --token is pretty important though so maybe want to add it until we can do the latter?

**Pull request merged**